server/start.go

```diff
@@ -88,9 +89,26 @@ const (
 	FlagMempoolMaxTxs = "mempool.max-txs"
 )
 
+// StartCmdOptions defines options that can be customized in `StartCmd`
+type StartCmdOptions struct {
+	AppCreator      types.AppCreator
+	DefaultNodeHome string
+	// DBOpener can be used to customize db opening, for example customize db options or support different db backends.
+	DBOpener func(rootDir string, backendType dbm.BackendType) (dbm.DB, error)
+	// PostSetup can be used to setup extra services under the same cancellable context,
+	// it's not called in stand-alone mode, only for in-process mode.
+	PostSetup func(svrCtx *Context, ctx context.Context, g *errgroup.Group) error
+	// AddFlags add custom flags to start cmd
+	AddFlags func(cmd *cobra.Command)
+}
+
 // StartCmd runs the service passed in, either stand-alone or in-process with
 // CometBFT.
-func StartCmd(appCreator types.AppCreator, defaultNodeHome string) *cobra.Command {
+func StartCmd(opts StartCmdOptions) *cobra.Command {
```

Comment:
Can we get an API breaking changelog as well for that?

Reply:
done